src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -1080,16 +1081,21 @@ private void writeLogTo(WriteMethod method, OutputStream os) throws IOException
 
     @Override public File getLogFile() {
         LOGGER.log(Level.WARNING, "Avoid calling getLogFile on " + this, new UnsupportedOperationException());
```

Comment:
Given the number of existing callers, should we only log the stack trace at FINE level?

Reply:
Well…we only _know_ about those callers because it is at `WARNING`. And this would be important e.g. for getting sentry.io to show violations in Evergreen.

Reply:
Isn't this potentially going to generate a log of logspam though?

Reply:
Also, I don't see that (at least on current Jenkins master branch) that Run.getLogFile is deprecated, which makes it feel very odd to me that we're deprecated its override in `WorkflowJob`

Reply:
> Isn't this potentially going to generate a log of logspam though?

Potentially. Note that the log call code was already there—not touched by this PR.

> I don't see that (at least on current Jenkins master branch) that Run.getLogFile is deprecated

Yup, on my to-do list.

Reply:
I think you might get away with marking it FINE or lower, but... as a user isn't it pretty obnoxious to see a bunch of log warnings due to a plugin using an API that isn't generally deprecated yet?

Reply:
Well, all warnings are obnoxious if (a) you look at them, (b) you are not in a position to fix them.

Anyway, again, that is not part of this PR. Can have a separate discussion about systems for limiting repeated log messages etc.

Reply:
>>I don't see that (at least on current Jenkins master branch) that Run.getLogFile is deprecated

>Yup, on my to-do list.

Deprecation should happen first in upstream and then followed in downstream. In this case, we are doing opposite? Whats the rationale for that? If at all, this warning should be coming from upstream, no?

Reply:
> first in upstream and then followed in downstream

In general, for JEP-210 the intention was to produce something that can run on existing LTS lines. There are various follow-up tasks. JEP-207 [discusses this particular topic](https://github.com/jenkinsci/jep/blob/master/jep/207/README.adoc#run-api-compatibility).

Reply:
What we should do is deprecate in Jenkins core (with an explanation in comments about why), log at a FINE level, and then when that releases in an LTS, we upgrade the log message to WARNING.  This also adds a window of time for any plugins that might be (ab)using getLogFile to change over before we starting making lots of logspam.

Reply:
Why?  While it may be satisfying to simple signal "stop using this API" by making it log a warning every time, for a *user* this looks like the Workflow Job plugin was broken by the update that added the logging - we've seen this before with "cosmetic" log warnings.  Just not great optics or user experience. 

Being able to point to deprecation of the extension point in core and an appropriate waiting period gives us more of a leg to stand on here.

Reply:
so what do I call instead?
I'm making changes to the build-failure-analyzer plugin, and saw my logs being spammed here.
but there's no javadoc explaining what the replacement / another solution is?

Reply:
@timja Any log-related methods other than `getLogFile`. `getLogText` is the basic one. In your case I would suggest simply

```diff
-String currentFile = build.getLogFile().getName();
+String currentFile = "log";
```

Reply:
Will give it a go later on thanks 👍

Reply:
> I don't see that (at least on current Jenkins master branch) that Run.getLogFile is deprecated

https://github.com/jenkinsci/jenkins/pull/3963